labapi/build.gradle

```diff
@@ -11,6 +17,45 @@ project.ext.vstsPassword = System.getenv("ENV_VSTS_MVN_ANDROIDCOMMON_ACCESSTOKEN
 //labapi version.  Let's not change this please
 version = getAppVersionName()
 
+def createTempUserApiClientCode = ""
+
+if (project.hasProperty("functionApiCreateTempUserCode")) {
+    createTempUserApiClientCode = functionApiCreateTempUserCode
+}
+
+def resetApiClientCode = ""
+
+if (project.hasProperty("functionApiResetCode")) {
+    resetApiClientCode = functionApiResetCode
+}
+
+def enablePolicyApiClientCode = ""
+
+if (project.hasProperty("functionApiEnablePolicyCode")) {
+    enablePolicyApiClientCode = functionApiEnablePolicyCode
+}
+
+def disablePolicyApiClientCode = ""
+
+if (project.hasProperty("functionApiDisablePolicyCode")) {
+    disablePolicyApiClientCode = functionApiDisablePolicyCode
+}
+
+buildConfig {
+    // forces the package. Defaults to '${project.group}'
+    packageName("com.microsoft.identity.internal.test.labapi")
+    useJavaOutput() // forces the outputType to 'java'
+}
+
+sourceSets {
+    main {
+        buildConfigField("String", "CREATE_TEMP_USER_API_CLIENT_CODE", "\"$createTempUserApiClientCode\"")
+        buildConfigField("String", "RESET_API_CLIENT_CODE", "\"$resetApiClientCode\"")
+        buildConfigField("String", "ENABLE_POLICY_API_CLIENT_CODE", "\"$enablePolicyApiClientCode\"")
+        buildConfigField("String", "DISABLE_POLICY_API_CLIENT_CODE", "\"$disablePolicyApiClientCode\"")
+    }
+}
```

Comment:
In what cases are these values going to be different,
If we don't need configurable values, we can rid of these flags

Reply:
I'm actually not sure what these codes represent but if these are secrets then can't be in source and so having variables or flags is probably the way to go.

Reply:
That's the issue, each of the new api function requires a different code. I don't like this implementation, but not sure we can do it any other way for now

Reply:
Updated this, no longer passed as build configs. instead we are using lab secret api to fetch them at runtime from the labs vault.